From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Thu, 25 Oct 2012 12:07:04 -0400
Subject: [PATCH] autosave: support RTEMS 4.10 mount()

RTEMS 4.10 provides a generic interface for mounting
file systems (including NFS).

The unmount() call remains unchanged.
---
 autosave/asApp/src/os/RTEMS/osdNfs.c |   12 +++++++++++-
 autosave/asApp/src/os/RTEMS/osdNfs.h |    4 ++++
 2 files changed, 15 insertions(+), 1 deletions(-)

diff --git a/autosave/asApp/src/os/RTEMS/osdNfs.c b/autosave/asApp/src/os/RTEMS/osdNfs.c
index 690d426..e23637d 100644
--- a/autosave/asApp/src/os/RTEMS/osdNfs.c
+++ b/autosave/asApp/src/os/RTEMS/osdNfs.c
@@ -41,7 +41,17 @@ int mountFileSystem(char *uidhost, char *addr, char *path, char *mntpoint)
     if (!mntpoint || !mntpoint[0]) return NFS_INVALID_MNTPOINT;
 
     /* mount the file system */
-    if (nfsMount(uidhost, path, mntpoint) == OK) {     /* 0 - succeed; -1 - failed */
+#ifdef RTEMS_FILESYSTEM_TYPE_NFS
+    /* RTEMS >=4.10 provides generic mount() call */
+    if (mount_and_make_target_path(uidhost, path,
+                                   RTEMS_FILESYSTEM_TYPE_NFS,
+                                   RTEMS_FILESYSTEM_READ_WRITE,
+                                   NULL) == OK)
+#else
+    /* For earlier use special call */
+    if (nfsMount(uidhost, path, mntpoint) == OK)     /* 0 - succeed; -1 - failed */
+#endif
+    {
         save_restoreNFSOK    = 1;
         save_restoreIoErrors = 0;                      /* clean the counter */
         return NFS_SUCCESS;
diff --git a/autosave/asApp/src/os/RTEMS/osdNfs.h b/autosave/asApp/src/os/RTEMS/osdNfs.h
index fa5be8e..7d90908 100644
--- a/autosave/asApp/src/os/RTEMS/osdNfs.h
+++ b/autosave/asApp/src/os/RTEMS/osdNfs.h
@@ -13,7 +13,11 @@
 /* header files for RTEMS */
 #include <rtems.h>
 #include <bsp.h>
+#include <rtems/libio.h>
+#ifndef RTEMS_FILESYSTEM_TYPE_NFS
+/* Needed for RTEMS < 4.10 */
 #include <librtemsNfs.h>
+#endif
 
 #define NFS_PATH_LEN 255                /* string length for NFS related path */
 
-- 
